File: backend/app/schemas/delivery_address.py
from pydantic import BaseModel, Field


class DeliveryAddressCreate(BaseModel):
    recipient_name: str = Field(min_length=1, max_length=120)
    phone: str = Field(pattern=r"^[0-9+ -]{8,20}$")
    address_line1: str = Field(min_length=3, max_length=250)
    address_line2: str | None = Field(default=None, max_length=250)
    landmark: str | None = Field(default=None, max_length=160)
    village_or_city: str = Field(min_length=2, max_length=120)
    district: str = Field(min_length=2, max_length=120)
    state: str = Field(min_length=2, max_length=120)
    postal_code: str = Field(pattern=r"^[0-9]{6}$")
    is_default: bool = False


class DeliveryAddressUpdate(BaseModel):
    recipient_name: str | None = Field(default=None, min_length=1, max_length=120)
    phone: str | None = Field(default=None, pattern=r"^[0-9+ -]{8,20}$")
    address_line1: str | None = Field(default=None, min_length=3, max_length=250)
    address_line2: str | None = Field(default=None, max_length=250)
    landmark: str | None = Field(default=None, max_length=160)
    village_or_city: str | None = Field(default=None, min_length=2, max_length=120)
    district: str | None = Field(default=None, min_length=2, max_length=120)
    state: str | None = Field(default=None, min_length=2, max_length=120)
    postal_code: str | None = Field(default=None, pattern=r"^[0-9]{6}$")
    is_default: bool | None = None

File: mobile/lib/features/cart/models/delivery_address.dart
class DeliveryAddress {
  const DeliveryAddress({
    required this.id,
    required this.recipientName,
    required this.phone,
    required this.addressLine1,
    required this.villageOrCity,
    required this.district,
    required this.state,
    required this.postalCode,
    required this.isDefault,
    this.addressLine2,
    this.landmark,
  });

  final String id,
      recipientName,
      phone,
      addressLine1,
      villageOrCity,
      district,
      state,
      postalCode;
  final String? addressLine2, landmark;
  final bool isDefault;

  factory DeliveryAddress.fromJson(Map<String, dynamic> json) =>
      DeliveryAddress(
        id: json['id'].toString(),
        recipientName: json['recipient_name'].toString(),
        phone: json['phone'].toString(),
        addressLine1: json['address_line1'].toString(),
        addressLine2: json['address_line2']?.toString(),
        landmark: json['landmark']?.toString(),
        villageOrCity: json['village_or_city'].toString(),
        district: json['district'].toString(),
        state: json['state'].toString(),
        postalCode: json['postal_code'].toString(),
        isDefault: json['is_default'] as bool? ?? false,
      );

  String get summary =>
      '$addressLine1, $villageOrCity, $district - $postalCode';
}
